flake.crane.nix

```diff
@@ -0,0 +1,461 @@
+
+  # Filter only files relevant to building the workspace
+  filterWorkspaceFiles = src: filterSrcWithRegexes [ "Cargo.lock" "Cargo.toml" ".cargo" ".cargo/.*" ".*/Cargo.toml" ".*\.rs" ".*\.html" ".*/proto/.*" ] src;
+
+  # Like `filterWorkspaceFiles` but with `./scripts/` and `./misc/test/` included
+  filterWorkspaceTestFiles = src: filterSrcWithRegexes [ "Cargo.lock" "Cargo.toml" ".cargo" ".cargo/.*" ".*/Cargo.toml" ".*\.rs" ".*\.html" ".*/proto/.*" "scripts/.*" "misc/test/.*" ] src;
+
+  filterSrcWithRegexes = regexes: src:
+    let
+      basePath = toString src + "/";
+    in
+    lib.cleanSourceWith {
+      filter = (path: type:
+        let
+          relPath = lib.removePrefix basePath (toString path);
+          includePath =
+            (type == "directory") ||
+            lib.any
+              (re: builtins.match re relPath != null)
+              regexes;
+        in
+        # uncomment to debug:
+          # builtins.trace "${relPath}: ${lib.boolToString includePath}"
+        includePath
+      );
+      inherit src;
+    };
+in
+rec {
+
+  cargo-llvm-cov = craneLib.buildPackage rec {
+    pname = "cargo-llvm-cov";
+    version = "0.4.14";
+    buildInputs = commonArgs.buildInputs;
+
+    src = pkgs.fetchCrate {
+      inherit pname version;
+      sha256 = "sha256-DY5eBSx/PSmKaG7I6scDEbyZQ5hknA/pfl0KjTNqZlo=";
+    };
+    doCheck = false;
+  };
+
+  commonArgsBase = {
+    pname = "fedimint-workspace";
+
+    buildInputs = with pkgs; [
+      clang
+      gcc
+      openssl
+      pkg-config
+      perl
+      pkgs.llvmPackages.bintools
+      rocksdb
+      protobuf
+    ] ++ lib.optionals stdenv.isDarwin [
+      libiconv
+      darwin.apple_sdk.frameworks.Security
+      zld
+    ] ++ lib.optionals (!(stdenv.isAarch64 || stdenv.isDarwin)) [
+      # mold is currently broken on ARM and MacOS
+      mold
+    ];
+
+    nativeBuildInputs = with pkgs; [
+      pkg-config
+      ts
+
+      # tests
+      (hiPrio pkgs.bashInteractive)
+      bc
+      bitcoind
+      clightning-dev
+      electrs
+      jq
+      lnd
+      netcat
+      perl
+      pkgs-kitman.esplora
+      procps
+      which
+    ];
+
+
+    # https://github.com/ipetkov/crane/issues/76#issuecomment-1296025495
+    installCargoArtifactsMode = "use-zstd";
+
+    LIBCLANG_PATH = "${pkgs.libclang.lib}/lib/";
+    ROCKSDB_LIB_DIR = "${pkgs.rocksdb}/lib/";
+    PROTOC = "${pkgs.protobuf}/bin/protoc";
+    PROTOC_INCLUDE = "${pkgs.protobuf}/include";
+    CI = "true";
+    HOME = "/tmp";
+  };
+
+  commonArgs = commonArgsBase // {
+    src = filterWorkspaceFiles ./.;
+  };
+
+  commonArgsDepsOnly = commonArgsBase // {
+    cargoVendorDir = craneLib.vendorCargoDeps {
+      src = filterWorkspaceFiles ./.;
+    };
+    # copy over the linker/ar wrapper scripts which by default would get
+    # stripped by crane
+    dummySrc = craneLib.mkDummySrc {
+      src = filterWorkspaceDepsBuildFiles ./.;
```

Comment:
unrelated to this PR, but

> ./.

This explains why I don't get cached deps (I have cloned it in ~/src/minimint)
https://nix.dev/anti-patterns/language#reproducibility-referencing-top-level-directory-with

Reply:
Oh, you mean from the cachix? Yeah, that would explain it. Will fix.

Reply:
https://github.com/fedimint/fedimint/pull/1934

Reply:
> This explains why I don't get cached deps (I have cloned it in ~/src/minimint)

Same, thx for the explanation @Maan2003, was always wondering :see_no_evil: